shared/entity/type.go

```diff
@@ -221,11 +221,8 @@ func (t Type) URL(projectName string, location string, pathArguments ...string)
 		u = u.WithQuery("project", projectName)
 	}
 
-	// Always set location if provided.
-	if location != "" {
-		u = u.WithQuery("target", location)
-	}
-
+	// Always set location if provided (empty or "none" locations are ignored).
+	u = u.Target(location)
```

Comment:
Does this affect any end-user API output or is this just internal?
Also if "(empty or "none" locations are ignored)" why is this needed?

Reply:
The `Target` method called here is what ignores empty or "none" locations. 

This change is so that when calling this method with e.g. `(api.StorageVolume).Location` doesn't add "target=none" to the URL if we are in standalone mode, or "target=" for remote pools.